File: project-2/two.hs
fibEvenSum :: Integer -> Integer -> Integer -> Integer
fibEvenSum prevPrev prev sum
  | curr > 4000000 = sum
  | curr `mod` 2 == 0  = fibEvenSum prev curr (sum + curr)
  | otherwise      = fibEvenSum prev curr sum
 where
  curr = prevPrev + prev


main = putStrLn $ show $ fibEvenSum 0 1 0

File: project-2/two.go
package main

import "fmt"

func main() {	
	var sum uint32 = 0
	var prevPrev uint32 = 0
	var prev uint32 = 1

	for {
		var curr uint32 = prevPrev + prev
		if curr > 4000000 {
			break
		}

		if curr % 2 == 0 {
			sum += curr
		}
		prevPrev = prev
		prev = curr
	}

	fmt.Println(sum)
}
